tools/generate_schema_wrapper.py

```diff
@@ -564,7 +565,7 @@ def generate_vegalite_channel_wrappers(
     imports = imports or [
         "from __future__ import annotations\n",
         "from typing import Any, overload, Sequence, List, Literal, Union, TYPE_CHECKING",
-        "import pandas as pd",
+        "from narwhals.dependencies import is_pandas_dataframe",
```

Comment:
```suggestion
        "from narwhals.dependencies import is_pandas_dataframe as _is_pandas_dataframe",
```
Avoiding adding to `__all__` (1/2)
https://github.com/vega/altair/commit/7ab395bd10732ecf7bf3c7ffd969ac2eedc20feb

Reply:
@MarcoGorelli there is quite a lot of implicit handling of imports in `altair`.

Certain things are [special-cased](https://github.com/dangotbanned/altair/blob/801cccc80e7764df3d46cd0db0343f095ee78d01/tools/update_init_file.py) like `typing`, but generally the safest option is just aliasing w/ `_` prefix. 

I think you'd need to run those changes locally first though to remove `is_pandas_dataframe` from `__all__`

Reply:
aah thanks, that stumped me 😄